### DISCUSSION TITLE
Actions | Drupal at your Fingertips

### DISCUSSION THREAD

**Discussion by giscus** [selwynpolit/d9book#129] [Q&A]
# Actions | Drupal at your Fingertips

Drupal at your Fingertips: A developers quick reference for Drupal 9 and 10

https://www.drupalatyourfingertips.com/actions

**Comment by kopeboy** [comment]
How to place these actions on a page as links or buttons, taking current path & query as arguments for triggering the execution?

**Reply by kopeboy** [reply]
Is this relevant? https://www.drupal.org/docs/drupal-apis/menu-api/providing-module-defined-local-actions

**Reply by kopeboy** [reply]
From my current research, probably the easiest way is using ECA Endpoint module: you can define your path for the specific _ECA Endpoint Response_ event and use query parameters in the following (ECA) actions & conditions. To define custom access permissions you add the _ECA Endpoint Access_ event and apply conditions like _Role of current user_ before the _Set access result_ action.
